src/DictList.elm

```diff
@@ -226,6 +227,17 @@ decodeArray keyMapper valueDecoder =
         (Json.Decode.list valueDecoder)
 
 
+{-| Decodes a JSON array into the DictList. You supply two decoders. Given an element
+of your JSON array, the first decoder should decode the key, and the second decoder
+should decode the value.
+-}
+decodeArray2 : Decoder comparable -> Decoder value -> Decoder (DictList comparable value)
+decodeArray2 keyDecoder valueDecoder =
+    Json.Decode.map2 (,) keyDecoder valueDecoder
+        |> Json.Decode.list
+        |> Json.Decode.map fromList
+
```

Comment:
@amitaibu This should do the trick for the case we were discussing. If you're in Elm 0.17, then you'd use `object2` instead of `map2`.